solr/core/src/java/org/apache/solr/handler/admin/api/GetSchemaFieldAPI.java

```diff
@@ -0,0 +1,255 @@
+    @JsonProperty("dynamicFields")
+    public Object dynamicFields;
+  }
+
+  @GET
+  @Path("/dynamicfields/{fieldName}")
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_ATOM_XML, BINARY_CONTENT_TYPE_V2})
+  @PermissionName(PermissionNameProvider.Name.SCHEMA_READ_PERM)
+  public SchemaGetDynamicFieldInfoResponse getDynamicFieldInfo(
+      @PathParam("fieldName") String fieldName) {
+    SchemaGetDynamicFieldInfoResponse response =
+        instantiateJerseyResponse(SchemaGetDynamicFieldInfoResponse.class);
+    final String realName = "dynamicfields";
+
+    SimpleOrderedMap<Object> dynamicFieldInfo =
+        retrieveFieldInfoOfType(realName, fieldName, params);
+    if (dynamicFieldInfo != null) {
+      response.dynamicFieldInfo = dynamicFieldInfo;
+      return response;
+    }
+    throw new SolrException(
+        SolrException.ErrorCode.NOT_FOUND, "No such path /" + realName + "/" + fieldName);
+  }
+
+  public static class SchemaGetDynamicFieldInfoResponse extends SolrJerseyResponse {
+    @JsonProperty("dynamicField")
+    public SimpleOrderedMap<?> dynamicFieldInfo;
+  }
+
+  @GET
+  @Path("/fieldtypes")
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML, BINARY_CONTENT_TYPE_V2})
+  @PermissionName(PermissionNameProvider.Name.SCHEMA_READ_PERM)
+  public SchemaListFieldTypesResponse listSchemaFieldTypes() {
+    SchemaListFieldTypesResponse response =
+        instantiateJerseyResponse(SchemaListFieldTypesResponse.class);
+    final String realName = "fieldtypes";
+
+    response.fieldTypes = listAllFieldsOfType(realName, params);
+
+    return response;
+  }
+
+  public static class SchemaListFieldTypesResponse extends SolrJerseyResponse {
+    @JsonProperty("fieldTypes")
+    public Object fieldTypes;
+  }
+
+  @GET
+  @Path("/fieldtypes/{fieldTypeName}")
+  @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_ATOM_XML, BINARY_CONTENT_TYPE_V2})
+  @PermissionName(PermissionNameProvider.Name.SCHEMA_READ_PERM)
+  public SchemaGetFieldTypeInfoResponse getFieldTypeInfo(
+      @PathParam("fieldTypeName") String fieldTypeName) {
+    SchemaGetFieldTypeInfoResponse response =
+        instantiateJerseyResponse(SchemaGetFieldTypeInfoResponse.class);
+
+    final String realName = "fieldtypes";
+
+    SimpleOrderedMap<Object> fieldTypeInfo =
+        retrieveFieldInfoOfType(realName, fieldTypeName, params);
+    if (fieldTypeInfo != null) {
+      response.fieldTypeInfo = fieldTypeInfo;
+      return response;
+    }
+    throw new SolrException(
+        SolrException.ErrorCode.NOT_FOUND, "No such path /" + realName + "/" + fieldTypeName);
+  }
+
+  public static class SchemaGetFieldTypeInfoResponse extends SolrJerseyResponse {
+    @JsonProperty("fieldType")
+    public SimpleOrderedMap<?> fieldTypeInfo;
+  }
+
+  private Object listAllFieldsOfType(String realName, SolrParams params) {
+    String camelCaseRealName = IndexSchema.nameMapping.get(realName);
```

Comment:
[0] Mostly thinking aloud here, but I wonder what parameters the schema code looks for in this `SolrParams` object.  Obviously there's 'meta' below, but there's probably others in `getNamedPropertyValues`.

The more explicit we can be about the accepted parameters, the easier this code will be to read.  And the nicer any sort of client code will be if we start doing automated client generation (see SOLR-16835).  If the number of parameters is relatively small it might be worth calling each param out explicitly in the API methods above, and then   packaging them up in a SolrParam to satisfy the IndexSchema call below.

Reply:
I was doing an experiment before this approach to try and pass all the necessary parameters, but I have found [here](https://github.com/apache/solr/blob/8089e6fc6a9d37299df9a63e5372ec5c235e4f9a/solr/core/src/java/org/apache/solr/schema/IndexSchema.java#L1643-L1647) that we would need at least 6 parameters. I considered it a bit too much to nicely handle so I went ahead and used the parameter injection instead. 
If you think 6 parameters is a viable option I can easily change it back to that approach. Let me know what you think!

Reply:
6 is a lot, 🤔 

Ultimately, I think there will be value in being as explicit as possible here.  But most of those benefits rely on JIRA tickets that haven't been finished yet (e.g. using these API definitions to [auto-generate client code](https://github.com/apache/solr/pull/1681)).

So, I'm happy to go either way here.  It'll set us up really nicely if we're more explicit with the API params.  But if you don't want the method signature to be noisy in the meantime, that also makes sense to me.

Reply:
I ended up adding a TODO comment to the ctor where we inject the SolrParams as a reminder to circle back to this and be more explicit.

But I'm happy to merge as-is for now